File: src/main/scala/Segment.scala
case class Point(name: String, x: Long, y: Long) {
}

case class Segment(f: Point, t: Point) {
  def distance = math.sqrt((f.x - t.x) * (f.x - t.x) + (f.y - t.y) * (f.y - t.y))
}

case class Path(segments: List[Segment]) {
  def distance = segments.map(_.distance).sum
  val stops:Set[Point] = segments.flatMap(s => Set(s.f, s.t)).toSet
}

File: javaslang/quickstart/src/main/java/Segment.java
public class Segment {
    public final Point origin;
    public final Point dest;

    public Segment(Point origin, Point dest) {
        this.origin = origin;
        this.dest = dest;
    }

    public double getDistance() {
        double deltaX = origin.x - dest.x;
        double deltaY = origin.y - dest.y;
        return Math.sqrt(deltaX * deltaX + deltaY * deltaY);
    }
}
